src/daft-core/src/python/datatype.rs

```diff
@@ -283,49 +273,51 @@ impl PyDataType {
         Ok(DataType::Python.into())
     }
 
-    pub fn to_arrow(&self, cast_tensor_type_for_ray: Option<bool>) -> PyResult<PyObject> {
-        Python::with_gil(|py| {
-            let pyarrow = py.import(pyo3::intern!(py, "pyarrow"))?;
-            let cast_tensor_to_ray_type = cast_tensor_type_for_ray.unwrap_or(false);
-            match (&self.dtype, cast_tensor_to_ray_type) {
-                (DataType::FixedShapeTensor(dtype, shape), false) => Ok(
-                    if py
-                        .import(pyo3::intern!(py, "daft.utils"))?
-                        .getattr(pyo3::intern!(py, "pyarrow_supports_fixed_shape_tensor"))?
-                        .call0()?
-                        .extract()?
-                    {
-                        pyarrow
-                            .getattr(pyo3::intern!(py, "fixed_shape_tensor"))?
-                            .call1((
-                                Self {
-                                    dtype: *dtype.clone(),
-                                }
-                                .to_arrow(None)?,
-                                pyo3::types::PyTuple::new(py, shape.clone()),
-                            ))?
-                            .to_object(py)
-                    } else {
-                        // Fall back to default Daft super extension representation if installed pyarrow doesn't have the
-                        // canonical tensor extension type.
-                        ffi::to_py_schema(&self.dtype.to_arrow()?, py, pyarrow)?
-                    },
-                ),
-                (DataType::FixedShapeTensor(dtype, shape), true) => Ok(py
-                    .import(pyo3::intern!(py, "ray.data.extensions"))?
-                    .getattr(pyo3::intern!(py, "ArrowTensorType"))?
-                    .call1((
-                        pyo3::types::PyTuple::new(py, shape.clone()),
-                        Self {
-                            dtype: *dtype.clone(),
-                        }
-                        .to_arrow(None)?,
-                    ))?
-                    .to_object(py)),
-                (_, _) => ffi::to_py_schema(&self.dtype.to_arrow()?, py, pyarrow)?
-                    .getattr(py, pyo3::intern!(py, "type")),
-            }
-        })
+    pub fn to_arrow(
+        &self,
+        py: Python,
+        cast_tensor_type_for_ray: Option<bool>,
+    ) -> PyResult<PyObject> {
```

Comment:
is this a bad diff? what's different

Reply:
Ah, this is a driveby! This was me noticing that we were redundantly capturing the GIL with `Python::with_gil(|py| ...)` when pyo3 methods already have the GIL, and the `py: Python` marker token can be transparently injected into the method by specifying it as an argument.